File: Group-Video/OpenVideoCall-iOS-Objective-C/OpenVideoCall/EncryptionType.m
//
//  EncryptionType.m
//  OpenVideoCall
//
//  Created by CavanSu on 24/11/2017.
//  Copyright © 2017 Agora. All rights reserved.
//

#import "EncryptionType.h"

@implementation EncryptionType
+ (NSString *)modeStringWithEncrypType:(EncrypType)type {
    switch (type) {
        case EncrypTypeXTS128: return @"aes-128-xts"; break;
        case EncrypTypeXTS256: return @"aes-256-xts"; break;
        default: return @""; break;
    }
}

+ (NSString *)descriptionWithEncrypType:(EncrypType)type {
    switch (type) {
        case EncrypTypeXTS128: return @"AES 128"; break;
        case EncrypTypeXTS256: return @"AES 256"; break;
        default: return @""; break;
    }
}

+ (NSArray *)encrypTypeArray {
    return @[@(EncrypTypeXTS128), @(EncrypTypeXTS256)];
}
@end

File: Group Video/OpenVideoCall-macOS/OpenVideoCall/EncryptionType.swift
//
//  EncryptionType.swift
//  OpenVideoCall
//
//  Created by GongYuhua on 16/7/27.
//  Copyright © 2016年 Agora. All rights reserved.
//


enum EncryptionType {
    case xts128, xts256
    
    static let allValue: [EncryptionType] = [.xts128, .xts256]
    
    func modeString() -> String {
        switch self {
        case .xts128: return "aes-128-xts"
        case .xts256: return "aes-256-xts"
        }
    }
    
    func description() -> String {
        switch self {
        case .xts128: return "AES 128"
        case .xts256: return "AES 256"
        }
    }
}
